plugins/mtd/fu-mtd-device.c

```diff
@@ -258,6 +327,12 @@ fu_mtd_device_write_firmware(FuDevice *device,
 static void
 fu_mtd_device_init(FuMtdDevice *self)
 {
+	fu_device_set_summary(FU_DEVICE(self), "Memory Technology Device");
+	fu_device_add_protocol(FU_DEVICE(self), "org.infradead.mtd");
+	fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_INTERNAL);
+	fu_device_add_flag(FU_DEVICE(self), FWUPD_DEVICE_FLAG_NEEDS_REBOOT);
+	fu_device_add_internal_flag(FU_DEVICE(self), FU_DEVICE_INTERNAL_FLAG_ENSURE_SEMVER);
```

Comment:
Are we sure we have semantic version everywhere?

Reply:
That's a good question. I'm pretty sure we don't want to support UBoot versions of things like `1.2.3-cafebeef` as then we'll be doing `strcmp`-style version comparisons and that's not really going to work when `1.12.3-beefcafe` comes along. At least with this we can say for a specific device *we're expecting number.number* and do some sensible version comparisons.

Reply:
I agree with you now; we can add this as a `Flags = ensure-semver` in the same place we set the firmware format.